File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // Differentiate between a valid entry and an invalid entry
    if (argc == 2)
    {
        // Use argv[1] value as key integer
        int k = atoi(argv[1]);
        // Set up "plaintext" prompt
        string pl;
        {
            pl = get_string("plaintext: ");
        }
        // Set up "ciphertext" segment
        printf("ciphertext: ");
        for (int i = 0, n = strlen(pl); i < n; i++)
        {
            char f = pl[i];
            // Ensure plaintext entry is alphabetical
            if (isalpha(f))
            {
                // Differentiate between uppercase and lowercase entries in order to preserve case
                if (isupper(f))
                {
                    // Use key to instigate code
                    // Go from ASCII to alphabetical index to ASCII
                    char C = (f - 65 + k) % 26 + 65;
                    // Print the ciphertext
                    printf("%c", C);
                }
                if (islower(f))
                {

                    // Use key to instigate code
                    // Go from ASCII to alphabetical index to ASCII
                    char c = (f - 97 + k) % 26 + 97;
                    // Print the ciphertext
                    printf("%c", c);
                }
            }
        }
        printf("\n");
        return 0;
    }
    else
    {
        printf("Usage: ./caesar k\n");
        // handle cases when argc doesn't equal 2
        return 1;
    }
}

File: caesar.py
from cs50 import get_string
from sys import argv, exit


def main():
    # Return an error message and exit if length of argv isn't 2
    if len(argv) != 2:
        print('Usage: python caesar.py key')
        exit(1)
    # Recognize key
    key = int(argv[1])
    # Return an error message and exit if key is negative
    if key < 0:
        print('Key needs to be a non-negative integer')
        exit(2)
    # Prompt user for plaintext
    plaintext = get_string('plaintext: ')
    # Encrypt plaintext and make ciphertext
    print('ciphertext: ', end='')
    for i in plaintext:
        # Maintain uppercase form for letters and apply key
        if i.isupper():
            print(chr(((ord(i) - ord('A') + key) % 26) + ord('A')), end='')
        # Maintain lowercase form for letters and apply key
        elif i.islower():
            print(chr(((ord(i) - ord('a') + key) % 26) + ord('a')), end='')
        # Return all other characters as they are
        else:
            print(i, end='')
    # Return line
    print()


if __name__ == '__main__':
    main()
